### PR TITLE
Added ValidateFunc for `sigsci_corp_list`

### PR DESCRIPTION
**Pull request by ponkio-o**
Hi, 

Added some ValidateFunc for `sigsci_corp_list`. 

- `name`
- `type`
- `description`

Also, added new function “validStringLength” in `lib.go`. It’s validate of string length. It's useful when you need to validate the string length.

## Why only validate length?
Since the WebUI displays validation rules when an invalid value is entered, I considered creating a function like "validRegex" that could validate these using regular expressions.
(For example: https://github.com/hashicorp/terraform-provider-aws/blob/3db3102b34357187208b87b6825b6dcfd65e4bbb/.ci/providerlint/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation/strings.go#L90-L106)
image

However, since the official documentation does not mention validation using regular expressions, I limited ourselves to length validation.
https://docs.fastly.com/signalsciences/api/#_corps__corpName__lists_post
image

### CODE REVIEW

File: provider/resource_corp_integration.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ func resourceCorpIntegration() *schema.Resource {
 				ForceNew:    true,
 				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
 					if !existsInString(val.(string), "mailingList", "slack", "microsoftTeams") {
-						return nil, []error{fmt.Errorf(`"received type %q is invalid. should be "mailingList", "slack", or "microsoftTeams"`, val.(string))}
+						return nil, []error{fmt.Errorf(`received type %q is invalid. should be "mailingList", "slack", or "microsoftTeams"`, val.(string))}
```
Comment (ponkio-o) [inline provider/resource_corp_integration.go:28]: I had missed in https://github.com/signalsciences/terraform-provider-sigsci/pull/149

File: provider/resource_corp_list.go [hunk 2]
```diff
@@ -22,17 +22,35 @@ func resourceCorpList() *schema.Resource {
...
+					return nil, nil
+				},
 			},
 			"description": {
 				Type:        schema.TypeString,
 				Description: "Optional list description",
 				Optional:    true,
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					if !validStringLength(val.(string), 0, 140) {
+						return nil, []error{fmt.Errorf(`received name %q is invalid. should be max len 140`, val.(string))}
```
Comment (shawnps) [inline provider/resource_corp_list.go:50]:
```suggestion
						return nil, []error{fmt.Errorf(`received description %q is invalid. should be max len 140`, val.(string))}
```

Comment (ponkio-o) [reply provider/resource_corp_list.go:50]: oh sorry! thank you for your suggestion.

**Review by shawnps — APPROVED** [review]
Thank you @ponkio-o!

**Pull request merged**